app/services/storage.py

```
@@ -22,9 +25,14 @@
     try:
         with open(settings.gcp_keypath, "r") as f:
             credentials_info = json.load(f)
+        client = storage.Client.from_service_account_info(credentials_info)
+        logging.info(f"GCS client initialized for project: {client.project}")
     except FileNotFoundError as exc:
+        logging.error(f"GCP key file not found: {settings.gcp_keypath}")
         raise FileNotFoundError(f"GCP key file not found: {settings.gcp_keypath}") from exc
-    client = storage.Client.from_service_account_info(credentials_info)
+    except Exception as e:
+        logging.error(f"Failed to initialize GCS client: {str(e)}")
+        raise RuntimeError(f"Failed to initialize GCS client: {str(e)}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)